guides/plugins/plugins/administration/add-rule-assignment-configuration.md

```diff
@@ -0,0 +1,256 @@
+
+This guide **does not** explain how to create a new plugin for Shopware 6.
+Head over to our Plugin base guide to learn how to create a plugin at first:
+
+{% page-ref page="../plugin-base-guide.md" %}
+
+## Creating the index.js file
+
+The first step is creating a new directory like so `<plugin root>/src/Resources/app/administration/src/module/sw-settings-rule/extension/sw-settings-rule-detail-assignments`.
+Right afterwards, create a new file called `index.js` in there.
+
+Your custom module directory isn't known to Shopware 6 yet.
+The entry point of your plugin is the `main.js` file.
+That's the file you need to change now, so that it loads your extended component.
+For this, simply add the following line to your `main.js` file:
+
+{% code title="<plugin root>/src/Resources/app/administration/src/main.js" %}
+```javascript
+import './module/sw-settings-rule/extension/sw-settings-rule-detail-assignments';
+```
+{% endcode %}
+
+Now your module's `index.js` will be executed.
+
+## Override the component
+
+Your `index.js` is still empty now, so let's override the `sw-settings-rule-detail-assignments` component.
+This is technically done by calling the method `override` method of our [ComponentFactory](https://github.com/shopware/platform/blob/trunk/src/Administration/Resources/app/administration/src/core/factory/component.factory.ts), which is available throughout our third party wrapper.
+This method expects a name, and a configuration for the component you want to override.
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-settings-rule/extension/sw-settings-rule-detail-assignments/index.js" %}
+```javascript
+Component.override('sw-settings-rule-detail-assignments', {
+    // override configuration here
+});
+```
+{% endcode %}
+
+## Overriding the computed
+
+Now your plugin is overriding the `sw-settings-rule-detail-assignments` component, but currently this has no effect.
+In the `associationEntitiesConfig` computed property the configuration of the rule assignment is built and returned to the method which initiate the component.
+Because of this, you have to override this computed property, get the computed property of the original component, add your own configuration of the rule assignment and return the whole configuration array.
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-settings-rule/extension/sw-settings-rule-detail-assignments/index.js" %}
+```javascript
+Component.override('sw-settings-rule-detail-assignments', {
+    computed: {
+        associationEntitiesConfig() {
+            const associationEntitiesConfig = this.$super('associationEntitiesConfig');
+            associationEntitiesConfig.push(/* insert your configuration here */);
+            return associationEntitiesConfig;
+        },
+    }
+});
+```
+{% endcode %}
+
+## Adding the configuration
+
+The configuration of the rule assignment is an object and offers a wide range of configuration options.
```

Comment:
The configuration of the rule assignment is passed as an object and offers a wide range of options.